core/dbt/adapters/factory.py

```diff
@@ -163,10 +164,22 @@ def get_adapter_type_names(self, name: Optional[str]) -> List[str]:
 
 def register_adapter(config: AdapterRequiredConfig) -> None:
     FACTORY.register_adapter(config)
-
-
-def get_adapter(config: AdapterRequiredConfig):
-    return FACTORY.lookup_adapter(config.credentials.type)
+    # Also register the python_adapter if it is specified
+    if config.python_adapter_credentials is not None:
+        FACTORY.register_adapter(config, config.python_adapter_credentials.type)
+
+
+def get_adapter(config: AdapterRequiredConfig, language: str = 'sql'):
+    if language == 'sql':
+        return FACTORY.lookup_adapter(config.credentials.type)
+    elif language == 'python':
+        # TODO: should we add `python_adapter_credentials` to AdapterRequiredConfig?
```

Comment:
This is a good question, and I am also not sure about the answer as well.

If any of the combined adapters (Python+SQL) change their initialization behaviour (e.g. instead of asking for permissions on `run_sql`, if they ask permission for `run_sql` + `run_python`) by default without running any Python models, it might make sense let them know that their Python computation functionality wont be used at all. Not sure if this is the case for the existing ones though. CC: @burkaygur @turbo1912